include/mra/kernels/multiply.h

```diff
@@ -11,64 +11,126 @@ namespace mra {
+        }
+
+
+        // fcube_for_mul(D, keyA, keyB.right_child(), nodeB, rcnodeB, phibar, phi, quad_x, K, workspace);
+        // T scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyB.right_child().level())));
+        // rcnodeB *= scale;
+        // fcube_for_mul(D, keyA, keyB.left_child(), nodeB, lcnodeB, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyB.left_child().level())));
+        // lcnodeB *= scale;
+        // fcube_for_mul(D, keyA, keyA.right_child(), nodeA, rcnodeA, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyA.right_child().level())));
+        // rcnodeA *= scale;
+        // fcube_for_mul(D, keyA, keyA.left_child(), nodeA, lcnodeA, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyA.left_child().child.level())));
+        // lcnodeA *= scale;
+      }
+      else if (keyB.level() <= keyB.level()){
+        T scale;
+        for (int i=0; i< keyA.num_children(); ++i){
+          fcube_for_mul(D, keyB.child_at(i), keyA.child_at(i), nodeA, cnodeA[i], phibar, phi, quad_x, K, workspace);
+          scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyB.right_child().level())));
+          cnodeB[i] *= scale;
+
+          fcube_for_mul(D, keyB.child_at[i], keyB.child_at(i), nodeB, cnodeB[i], phibar, phi, quad_x, K, workspace);
+          scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyA.right_child().level())));
+          cnodeA[i] *= scale;
+        }
+
+        // fcube_for_mul(D, keyB, keyA.right_child(), nodeA, rcnodeA, phibar, phi, quad_x, K, workspace);
+        // T scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyB.right_child().level())));
+        // rcnodeA *= scale;
+        // fcube_for_mul(D, keyB, keyA.left_child(), nodeA, lcnodeA, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyB.left_child().level())));
+        // lcnodeA *= scale;
+        // fcube_for_mul(D, keyB, keyB.right_child(), nodeB, rcnodeB, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyA.right_child().level())));
+        // rcnodeB *= scale;
+        // fcube_for_mul(D, keyB, keyB.left_child(), nodeB, lcnodeB, phibar, phi, quad_x, K, workspace);
+        // scale = std::sqrt(D.template get_volume<T>()*std::pow(T(0.5), T(NDIM*keyA.left_child().child.level())));
+        // lcnodeB *= scale;
+
+        // fcube_for_mul() returns function values evaluated at quadrature points
+      }
       foreach_idx(nodeA, [&](size_type i) {
-          r[i] = scale * r1[i] * r2[i];
+        foreach_idx(cnodeA[i], [&](size_type j) {
+          cnodeA[i][j] = cnodeA[i][j] * cnodeB[i][j];
       });
```

Comment:
We cannot nest two foreach_idx. Make the outer idx a regular for loop that iterates over the number of children.

Reply:
Gotcha! Will fix it in next commit